src/programs.rkt

```diff
@@ -123,6 +123,12 @@
     ['fl (λ (repr x) (real->repr x repr))]
     ['ival (λ (repr x) (if (ival? x) x (mk-ival (->bf x repr))))]
     ['nonffi (λ (repr x) x)]))
+
+  (define precision->real (match mode
+    ['bf identity]
+    ['fl (curryr repr->real repr)]
+    ['ival identity]
+    ['nonffi identity]))
```

Comment:
Why do we want the output of `eval-prog?` to be real?

Reply:
Currently changing it back to repr. This was the cause of the float.rkt mess.